OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -482,5 +482,17 @@ public static void AddPlayerFlagAndName(ScrollItemWidget template, Player player
 			};
 			playerName.GetColor = () => player.Color.RGB;
 		}
+
+		public static string GetExternalIP(int clientIndex, OrderManager orderManager)
+		{
+			var address = orderManager.LobbyInfo.ClientWithIndex(clientIndex).IpAddress;
+			if (clientIndex == orderManager.LocalClient.Index && address == IPAddress.Loopback.ToString())
+			{
+				if (UPnP.NatDevice != null && UPnP.NatDevice.GetExternalIP().ToString() != null)
```

Comment:
This is strange. When I call .ToString() on a null IPAddress it crashes like in https://github.com/OpenRA/OpenRA/issues/8787:

![image](https://cloud.githubusercontent.com/assets/756669/9703386/239facea-5483-11e5-9e1f-7383b955e575.png)